src/app/functions/server/refreshStoredScanResults.ts

```diff
@@ -18,59 +19,26 @@ import { logger } from "./logging";
  * @param profileId {number} OneRep Profile ID to refresh.
  */
 export async function refreshStoredScanResults(profileId: number) {
-  // This contains the latest scan results in our database.
-  const latestScan = await getLatestOnerepScanResults(profileId);
-
-  let scanId = latestScan.scan?.onerep_scan_id;
-  let reason = latestScan.scan?.onerep_scan_reason;
-  let status = latestScan.scan?.onerep_scan_status;
-
   try {
-    const currentScan = await getAllScanResults(profileId);
-    let newScanResults: ScanResult[] = [];
-
-    if (!latestScan.results.length) {
-      newScanResults = currentScan;
-    } else {
-      currentScan.forEach((current) => {
-        latestScan.results.forEach((latest) => {
-          if (
-            current.id === latest.onerep_scan_result_id &&
-            current.scan_id === latest.onerep_scan_id &&
-            new Date(current.updated_at) > latest.updated_at
-          ) {
-            newScanResults.push(current);
-          }
-        });
-      });
+    const remoteScans = (await listScans(profileId)).data;
+    const localScans = await getAllScansForProfile(profileId);
+
+    const newScans = remoteScans.filter(
+      (remoteScan) =>
+        !localScans.some(
+          (localScan) => localScan.onerep_scan_id === remoteScan.id,
+        ),
+    );
+
+    // Record any new scans.
+    for (const scan of newScans) {
+      await setOnerepScan(profileId, scan.id, scan.status, scan.reason);
```

Comment:
As I understand it, this inserts the scans sequentially (i.e. `await`ing the previous insert before starting the next one). Might be better to run the inserts in parallel:

```suggestion
    // Record any new scans.
    await Promise.all(newScans.map(scan => {
      await setOnerepScan(profileId, scan.id, scan.status, scan.reason);
    }));
```

Reply:
sgtm, committed with a small change:
```
    // Record any new scans.
    await Promise.all(
      newScans.map(async (scan) => {
        await setOnerepScan(profileId, scan.id, scan.status, scan.reason);
      }),
    );
```